src/Microsoft.AspNetCore.Mvc.Razor/Internal/ExpressionRewriter.cs

```diff
@@ -137,49 +137,38 @@ public override SyntaxNode VisitSimpleLambdaExpression(SimpleLambdaExpressionSyn
             // We want to make the new span
             var originalSpan = node.GetLocation().GetMappedLineSpan();
 
-            // Start by collecting all the trivia 'inside' the expression - we need to tack that on the end, but
-            // if it ends with a newline, don't include that.
-            var innerTrivia = SyntaxFactory.TriviaList(node.DescendantTrivia(descendIntoChildren: n => true));
-            if (innerTrivia.Count > 0 && innerTrivia[innerTrivia.Count - 1].IsKind(SyntaxKind.EndOfLineTrivia))
+            var charactersToExclude = memberAccess.Identifier.Text.Length;
+            var triviaList = new SyntaxTriviaList();
+
+            // Go through each token and
+            // 1. Append leading trivia
+            // 2. Append the same number of whitespace as the length of the token text
+            // 3. Append trailing trivia
+            foreach (var token in node.DescendantTokens())
             {
-                innerTrivia = innerTrivia.RemoveAt(innerTrivia.Count - 1);
-            }
+                if (token.HasLeadingTrivia)
+                {
+                    triviaList = triviaList.AddRange(token.LeadingTrivia);
+                }
 
-            memberAccess = memberAccess.WithTrailingTrivia(innerTrivia);
+                var padding = token.Text.Length;
+                // Need to exclude the length of the member name from the padding.
+                while (charactersToExclude > 0 && padding > 0)
+                {
+                    padding--;
+                    charactersToExclude--;
+                }
```

Comment:
It is not straightforward. `Math.Min(padding, 0)` will always return 0. I can change this while to 
```C#
if (padding > charactersToExclude)
{
    padding -= charactersToExclude;
    charactersToExclude = 0;
}
else
{
    charactersToExclude -= padding;
    padding = 0;
}
```
But I think they are both equally readable.